README.md

```diff
@@ -0,0 +1,695 @@
+        * [Block Comments](#block_comments)
+        * [Inline Comments](#inline_comments)
+    * [Naming Conventions](#naming_conventions)
+    * [Objects](#objects)
+    * [Functions](#functions)
+    * [Strings](#strings)
+    * [Conditionals](#conditionals)
+    * [Looping and Comprehensions](#looping_and_comprehensions)
+    * [Extending Native Objects](#extending_native_objects)
+    * [Exceptions](#exceptions)
+    * [Annotations](#annotations)
+    * [Miscellaneous](#miscellaneous)
+
+<a name="code_layout"></a>
+## Code layout
+
+<a name="tabs_or_spaces"></a>
+### Tabs or Spaces?
+
+Use **spaces only**, with **2 spaces** per indentation level. Never mix tabs and spaces.
+```
+'no_tabs': {
+  'level': 'error'
+}
+```
+
+Indentation is critical to Coffeescript and is set to use 2 spaces
+```
+'indentation': {
+  'value': 2
+}
+```
+
+<a name="maximum_line_length"></a>
+### Maximum Line Length
+
+Limit all lines to a maximum of 120 characters.
+```
+'max_line_length': {
+  'value': 120,
+  'level': 'warn',
+  'limitComments': true
+}
+```
+
+<a name="blank_lines"></a>
+### Blank Lines
+
+Separate top-level function and class definitions with a single blank line.
+```
+'newlines_after_classes': {
+  'level': 1,
+  'level': 'error'
+}
+```
+
+Separate method definitions inside of a class with a single blank line.
+
+Use a single blank line within the bodies of methods or functions in cases where this improves readability (e.g., for the purpose of delineating logical sections).
+
+<a name="optional_commas"></a>
+### Optional Commas
+
+Always use commas where they would be used in JavaScript
+
+```coffeescript
+# Yes
+foo = [
+  'some',
+  'string',
+  'values'
+]
+bar:
+  label: 'test',
+  value: 87
+
+# No
+foo = [
+  'some'
+  'string'
+  'values'
+]
+bar:
+  label: 'test'
+  value: 87
+```
+
+<a name="semicolons"></a>
+### Semicolons
+
+No trailing semicolons
+```
+'no_trailing_semicolons': {
+  'level': 'error'
+}
+```
+
+```coffeescript
+   console.log('a') # Yes
+   console.log('b'); # No
+```
+
+<a name="end_newline"></a>
+### End files on a newline
+['eol_last': {
+  'level': 'error'
+}]
+
+<a name="encoding"></a>
+### Encoding
+
+UTF-8 is the preferred source file encoding.
+
+<a name="module_imports"></a>
+## Module Imports
+
+If using a module system (CommonJS Modules, AMD, etc.), `require` statements should be placed on separate lines.
+
+```coffeescript
+require 'lib/setup'
+Backbone = require 'backbone'
+```
+These statements should be grouped in the following order:
+
+1. Standard library imports _(if a standard library exists)_
+2. Third party library imports
+3. Local imports _(imports specific to this application or library)_
+
+<a name="whitespace"></a>
+## Arrows
+Include spaces before and after an arrow _unless_ there are no arguments passed to the function in which case the preceding space can be omitted
+```
+'arrow_spacing': {
+  'level': 'error'
+}
+```
+```coffeescript
+    # Yes
+    x(-> 3)
+    x( -> 3)
```

Comment:
I think this should maybe be under `#No`, because the extra space is unnecessary and makes the arrow function feel unbalanced here.